keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/Resource.kt

```diff
@@ -13,6 +13,18 @@ data class Resource<out T : ResourceSpec>(
     require(metadata["application"].isValidApplication()) { "application must be a valid application" }
   }
 
+  val id: String
+    get() = metadata.getValue("id").toString()
+
+  val serviceAccount: String
+    get() = metadata.getValue("serviceAccount").toString()
+
+  val application: String
+    get() = metadata.getValue("application").toString()
+
+  val environment: String
```

Comment:
This is new. I added it as it's needed to find things like notification config in a `ResourceHandler`.